### ISSUE TITLE
Carousal: the scrollable element <div> with non-interactive content is not tabbable

### ISSUE THREAD

**Issue by amal-k-joy** [carbon-design-system/ibm-products#7305]
The carousal container has overflow: scroll , but the tab index is not set causing this accessibility violation. 

This issue is observed in interstitial and coachmarkOverlayElements

![Image](https://github.com/user-attachments/assets/e9f870e6-c41d-419b-a18b-7ccca6723145)

**Comment by amal-k-joy** [comment]
Looks like @makafsal fix on [this one](https://github.com/carbon-design-system/ibm-products/pull/7403) should solve this issue.

@elycheea any thoughts?

**Comment by jlongshore** [comment]
[Slack conversation](https://ibm-studios.slack.com/archives/C036P1CTN/p1744116605709549) - closing as not planned... A11y rule checker will be updated to reflect this.
